### DISCUSSION TITLE
How to create a list of non upgradable software for winget

### DISCUSSION THREAD

**Discussion by Braedach** [alkampfergit/personal-blog#11] [General]
In reference to your code here

https://www.codewrecks.com/post/general/winget-update-selective/

Your code does not allow for a null value - ie.. nothing to upgrade

I have put your code in the following if statement after you call the winget upgrade

if (!($upgradeResult -match "No installed package found matching input criteria."))
{
your code
}

Now for the proactive remediation bit in MEM

Love your work.

**Comment by d0verenny** [comment]
I receive an error:
![image](https://user-images.githubusercontent.com/28781690/194718613-8c3f071a-8c65-43f2-a60a-b2c9608b3cfb.png)
 
 Anyone had it too?

**Reply by alkampfergit** [reply]
It seems that windows deployment kit have a version that starts with char '<' it make the code that parses version to crash ... curious....

Probably you need simply to trim version string for space and < chars.

![image](https://user-images.githubusercontent.com/358545/195804167-7572d80c-5b9e-41c7-a551-3b0244221af3.png)

**Reply by d0verenny** [reply]
Hi! Thank you for the reply! Could you please elaborate, where do I trim it? Please note that I only excluded Unity from upgrades in the ps.1 script.  I need the rest of the stuff to be upgraded.

**Reply by d0verenny** [reply]
I found the root of the problem. I had a couple of different Windows SDK versions installed together. Probably, some other program installed it as a prerequisite before, and I forgot about it. I just uninstalled both the versions via usual "Add or remove programs" in Windows, and reinstalled WindowsSDK via winget again. The error is gone now.

**Comment by alkampfergit** [comment]
Perfect!!

Il sab 15 ott 2022, 11:02 maksimilinskii ***@***.***> ha
scritto:

**Comment by alienryes** [comment]
Great script, thanks! At present it doesn't account for packages that have versions set to unknown (e.g. Speedfan). So I changed the get line to: 

    $upgradeResult = winget upgrade --include-unknown | Out-String

and then the filter line to:

    if (-not (($toSkip -contains $package.Id) -or ($package.Version -like "unknown")))
